### PR TITLE
sccache: add dist-server build

### PR DESCRIPTION
**Pull request by viric**
###### Motivation for this change

Have sccache dist-server available, as its "--features=all" didn't bring it.
Following: https://github.com/mozilla/sccache/blob/master/docs/DistributedQuickstart.md

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/sccache/default.nix [hunk 1]
```diff
@@ -13,7 +13,7 @@ rustPlatform.buildRustPackage rec {
 
   cargoSha256 = "1cfdwf00jgwsv0f72427asid1xr57s56jk5xj489dgppvgy7wdbj";
 
-  cargoBuildFlags = [ "--features=all" ];
+  cargoBuildFlagsArray = [ "--features=dist-client dist-server" ];
```
Comment (zowoq) [inline pkgs/development/tools/misc/sccache/default.nix:16]: Comma separated is how this is handled currently.

```suggestion
  cargoBuildFlags = [ "--features=dist-client,dist-server" ];
```

Comment (viric) [reply pkgs/development/tools/misc/sccache/default.nix:16]: aaaaaah. So no need for spaces ever?

Comment (zowoq) [reply pkgs/development/tools/misc/sccache/default.nix:16]: Probably not, the [cargo build docs](https://doc.rust-lang.org/cargo/commands/cargo-build.html#feature-selection) says both are valid.

Comment (viric) [reply pkgs/development/tools/misc/sccache/default.nix:16]: reworked. thank you!

**Review by doronbehar — APPROVED** [review]

**Review by danieldk — CHANGES_REQUESTED** [review]
Could you update the commit message to follow the standard layout?

Looks good otherwise.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 115320` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>sccache: <a href="https://termbin.com/1udo">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/bf41882db4b5608544ccdb4bafaea9fd/raw/index.html">pretty log </a></li>
  </ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
sccache:
<details>

warning: EvalError
Cannot evaluate attribute ‘sccache’ in ‘/nix/store/r52yc2n84lbmpnr2iiak86mlrclsc1sf-nixpkgs’.
</details>
</details>

**Comment by danieldk** [comment]
> Result of `nixpkgs-review pr 115320` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

Seems like this is only supported on x86_64-linux anyway: https://github.com/mozilla/sccache/blob/b3f4727eda81c45b0632d036395af85429c4774b/src/bin/sccache-dist/main.rs#L70

I have added a commit to update `platforms`.

**Review by danieldk — APPROVED** [review]

**Pull request merged**